mm2src/coins/nft.rs

```diff
@@ -465,26 +658,51 @@ fn check_moralis_ipfs_bafy(token_uri: Option<&str>) -> Option<String> {
     })
 }
 
-async fn get_uri_meta(token_uri: Option<&str>, metadata: Option<&str>) -> UriMeta {
+async fn get_uri_meta(token_uri: Option<&str>, metadata: Option<&str>, url_antispam: &Url) -> UriMeta {
     let mut uri_meta = UriMeta::default();
+    // Fetching data from the URL if token_uri is provided
     if let Some(token_uri) = token_uri {
-        if let Ok(response_meta) = send_request_to_uri(token_uri).await {
-            if let Ok(token_uri_meta) = serde_json::from_value(response_meta) {
-                uri_meta = token_uri_meta;
-            }
+        if let Some(url) = construct_camo_url_with_token(token_uri, url_antispam) {
+            uri_meta = fetch_meta_from_url(url).await.unwrap_or_default();
         }
     }
+    // Filling fields from metadata if provided
     if let Some(metadata) = metadata {
         if let Ok(meta_from_meta) = serde_json::from_str::<UriMeta>(metadata) {
-            uri_meta.try_to_fill_missing_fields_from(meta_from_meta)
+            uri_meta.try_to_fill_missing_fields_from(meta_from_meta);
         }
     }
-    uri_meta.image_url = check_moralis_ipfs_bafy(uri_meta.image_url.as_deref());
-    uri_meta.animation_url = check_moralis_ipfs_bafy(uri_meta.animation_url.as_deref());
+    update_uri_moralis_ipfs_fields(&mut uri_meta);
     drop_mutability!(uri_meta);
     uri_meta
 }
 
+fn construct_camo_url_with_token(token_uri: &str, url_antispam: &Url) -> Option<Url> {
+    let mut url = url_antispam.clone();
+    url.set_path("url/decode");
+    let hex_token_uri = hex::encode(token_uri);
+    if let Ok(mut segments) = url.path_segments_mut() {
+        segments.push(hex_token_uri.as_str());
+    } else {
+        return None;
+    }
+    Some(url)
```

Comment:
You should be able to return `url` in the if block right away I think

Reply:
If we return url in if block there will be lifetime error as `path_segments_mut()` takes `&mut self` of url, we cant move it out, while segments is alive.

well if you dont like this `if else`, i can do  this
```rust
fn construct_camo_url_with_token(token_uri: &str, url_antispam: &Url) -> Option<Url> {
    if url_antispam.cannot_be_a_base() {
        return None;
    }
    let mut url = url_antispam.clone();
    url.set_path("url/decode");
    let hex_token_uri = hex::encode(token_uri);
    if let Ok(mut segments) = url.path_segments_mut() {
        segments.push(&hex_token_uri);
    }
    Some(url)
}
```

`path_segments_mut()` looks like this in source code. 
```rust
#[allow(clippy::result_unit_err)]
    pub fn path_segments_mut(&mut self) -> Result<PathSegmentsMut<'_>, ()> {
        if self.cannot_be_a_base() {
            Err(())
        } else {
            Ok(path_segments::new(self))
        }
    }
```
it propagates error if `url` cant be base. we can do `cannot_be_a_base` check our self at the beginning. this way we can remove  `else {return None}`. well it also allows us to delay the clone operation.

Reply:
UPD: oh I see, my suggestion checks just `url_antispam` as a base, but `path_segments_mut` checks `url` after we have set the path, lets keep `else {return None}`

Reply:
`hex_token_uri` can be moved inside the `Ok` clause to avoid its creation regardless.